src/main/java/payapal/Perfection.java

```diff
@@ -0,0 +1,42 @@
+package main.java.payapal;
+
+public class Perfection {
+
+    private static Perfection perf;
+
+    public synchronized static Perfection getPerf() {
+        if (perf == null) {
+            perf = new Perfection();
+        }
+        return perf;
+    }
+
```

Comment:
Here to avoid invalid state object creation, it's more likely to define a private constructor, or use @NoArgsConstructor with AccessLevel.PRIVATE.